src/TypeScript/EslintBridgeClient/EslintBridgeStartUp.cs

```diff
@@ -19,17 +19,183 @@
  */
 
 using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Abstractions;
+using System.Text.RegularExpressions;
 using System.Threading.Tasks;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Helpers;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator;
 
 namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient
 {
-    // TODO: https://github.com/SonarSource/sonarlint-visualstudio/issues/2256
-
     internal interface IEslintBridgeStartUp : IDisposable
     {
         /// <summary>
         /// Ensures that the process is running. Returns the port that server is running on.
         /// </summary>
         Task<int> Start();
     }
+
+    [Export(typeof(IEslintBridgeStartUp))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class EslintBridgeStartUp : IEslintBridgeStartUp
```

Comment:
The class name doesn't really convey what the class is doing; it's a wrapper that persists for the lifetime of the process so it can monitor the process and capture the output.

The class/interface name suggests that it just starts the process, which makes it non-obvious why it implements `IDisposable`, or that disposing the class will terminate the external process.

Reply:
I took the inspiration from the Owin classes, where you have a StartUp class that wraps the running app and is responsible for terminating it.
What do you think would be a better name? IEslintBridgeHost? IEslintBridgeProcess?

Reply:
Yeugh - that's not a helpful naming convention.
Either of those names would be better, and "process" is probably more accurate.

> I wanted to avoid giving away the fact that it's a process
Any specific reason? On the one hand it's an implementation detail, but it's a pretty important one that conveys useful information about what to expect about the behaviour of the class.

Reply:
Yep, just that it's an implementation detail. Renamed to `IEslintBridgeProcess`